vllm/core/block_manager.py

```diff
@@ -328,3 +459,41 @@ def get_num_free_gpu_blocks(self) -> int:
 
     def get_num_free_cpu_blocks(self) -> int:
         return self.cpu_allocator.get_num_free_blocks()
+
+    def access_all_blocks_in_seq(
+        self,
+        seq: Sequence,
+        access_time: float,
+    ) -> None:
+        block_table = self.block_tables[seq.seq_id]
+        for block in block_table:
+            block.last_accessed = access_time
+
+    def compute_all_blocks_in_seq(self, seq: Sequence):
+        if seq.seq_id not in self.block_tables:
+            return
+        block_table = self.block_tables[seq.seq_id]
+        for block in block_table:
+            block.computed = True
+
+    def get_all_computed_block_ids_seq(self, seq: Sequence) -> List[int]:
+        if seq.seq_id not in self.block_tables:
+            return []
+        block_table = self.block_tables[seq.seq_id]
+        # We want to get the first n contiguous completed blocks
+        return [
+            block.block_number
+            for block in takewhile(lambda block: block.computed, block_table)
+        ]
+
+    def get_all_computed_block_ids(self,
+                                   seq_group: SequenceGroup) -> List[int]:
+        ids_list = [
+            self.get_all_computed_block_ids_seq(seq)
+            for seq in iter(seq_group.seqs_dict.values())
```

Comment:
why is `iter` needed here? if you're just going through the list, it should work without an explicit iterator